src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/resources/RuntimeHelper.java.template

```diff
@@ -75,37 +75,34 @@ final class RuntimeHelper {
     }
 
     static final <Z> MemorySegment upcallStub(Class<Z> fi, Z z, FunctionDescriptor fdesc, String mtypeDesc) {
+        return upcallStub(fi, z, fdesc, mtypeDesc, ResourceScope.newImplicitScope());
+    }
+
+    static final <Z> MemorySegment upcallStub(Class<Z> fi, Z z, FunctionDescriptor fdesc, String mtypeDesc, NativeScope scope) {
```

Comment:
This reminds me, perhaps at some point we need to do another pass and get rid of NativeScope